feat(kafka sink): Add support for templates in the topic config

bruceg opened the pull request:
Closes #2385 

Signed-off-by: Bruce Guenter <bruce@timber.io>

Commit:
feat(kafka sink): Add support for templates in the topic config

Signed-off-by: Bruce Guenter <bruce@timber.io>

binarylogic reviewed (approved):
Nice work!

lukesteensen reviewed (approved):
Looks good! Would be nice to have a test that exercises this, but probably not a blocker.

Commit:
Merge remote-tracking branch 'origin/master' into kafka-topic-template

Signed-off-by: Bruce Guenter <bruce@timber.io>

Commit:
Interpret topic template in healthcheck

Signed-off-by: Bruce Guenter <bruce@timber.io>

Commit:
Use a topic template in the tests

Signed-off-by: Bruce Guenter <bruce@timber.io>

lukesteensen commented on src/sinks/kafka.rs:
```diff
@@ -210,13 +214,23 @@ impl Sink for KafkaSink {
 
 fn healthcheck(config: KafkaSinkConfig) -> super::Healthcheck {
     let client = config.to_rdkafka().unwrap();
+    let topic = match Template::from(config.topic).render_string(&Event::from("")) {
```
Instead of rendering an empty event, it might be better to check `Template::is_dynamic` and then just pass `None` to `fetch_metadata` below if it is.. That way we can still check that we can connect to the cluster.

bruceg replied:
My thinking was that template items like a date stamp could still be rendered and be able to do the full health check. However, I can do what you suggest for the metadata fetch part.

lukesteensen replied:
Yeah, that's fine with me!

Commit:
Actually do a health check when the template doesn't render

Just pass a `None` to the metadata fetch part instead of returning
`future::ok()`.

Signed-off-by: Bruce Guenter <bruce@timber.io>
